cumulusci/core/config/OrgConfig.py

```diff
@@ -178,3 +175,63 @@ def get_community_info(self, community_name, force_refresh=False):
             )
 
         return self._community_info_cache[community_name]
+
+    def has_minimum_package_version(self, package_identifier, version_identifier):
+        """Return True if the org has a version of the specified package that is
+        equal to or newer than the supplied version identifier.
+
+        The package identifier may be either a namespace or a 033 package Id.
+        The version identifier should be in "1.2.3" or "1.2.3b4" format.
+
+        A CumulusCIException will be thrown if you request to check a namespace
+        and multiple second-generation packages sharing that namespace are installed.
+        Use a package Id to handle this circumstance."""
+        installed_version = self.installed_packages.get(package_identifier)
+
+        if len(installed_version) > 1:
+            raise CumulusCIException(
+                f"Cannot check installed version of {package_identifier}, because multiple "
+                f"packages are installed that match this identifier."
+            )
+        elif not installed_version:
+            return False
+
+        return installed_version[0] >= version_identifier
+
+    @property
+    def installed_packages(self):
+        """installed_packages is a dict mapping a namespace or package Id (033*) to the installed package
+        version(s) matching that identifier. All values are lists, because multiple second-generation
+        packages may be installed with the same namespace.
+
+        To check if a required package is present, call `has_minimum_package_version()` with either the
+        namespace or 033 Id of the desired package and its version, in 1.2.3 format.
+
+        Beta version of a package are represented as "1.2.3b5", where 5 is the build number."""
+        if not self._installed_packages:
+            response = self.salesforce_client.restful(
+                "tooling/query/?q=SELECT SubscriberPackage.NamespacePrefix, SubscriSubscriberPackageVersion.MajorVersion, "
+                "SubscriberPackageVersion.MinorVersion, SubscriberPackageVersion.PatchVersion,  "
+                "SubscriberPackageVersion.BuildNumber, SubscriberPackageVersion.IsBeta "
+                "FROM InstalledSubscriberPackage"
+            )
+
+            self._installed_packages = defaultdict(lambda: [])
+            for package in response["records"]:
+                sp = package["SubscriberPackage"]
+                spv = package["SubscriberPackageVersion"]
+                # PatchVersion is a 0 on a non-patch version.
+                version = (
+                    f"{spv['MajorVersion']}.{spv['MinorVersion']}.{spv['PatchVersion']}"
+                )
+                if spv["IsBeta"]:
+                    version += f"b{spv['BuildNumber']}"
+                self._installed_packages[sp["NamespacePrefix"]].append(
+                    StrictVersion(version)
+                )
+                self._installed_packages[sp["Id"]].append(StrictVersion(version))
```

Comment:
Fortunately namespaces have to start with a letter so we don't have to worry about someone who tried to set their namespace to the 033 id!